framework/meta/src/cmd/contract/meta_config.rs

```diff
@@ -159,3 +189,62 @@ fn copy_to_wasm_unmanaged_ei() {
         fs::copy(WASM_LIB_PATH, WASM_NO_MANAGED_EI_LIB_PATH).unwrap();
     }
 }
+
+mod tests {
+    #[cfg(test)]
+    use crate::cmd::contract::sc_config::ContractVariantProfile;
+
+    #[test]
+    fn test_generate_cargo() {
+        let wasm_cargo_toml_data = super::WasmCargoTomlData::from(
+            "test".to_string(),
+            "2021".to_string(),
+            ContractVariantProfile::default(),
+            "0.44.0".to_string(),
+            Option::Some("../../../framework/base".to_string()),
+            Vec::<String>::new(),
+        );
+        let crate_name = "test-crate-name".to_string();
+        let generated_contents =
+            super::generate_wasm_cargo_toml(&wasm_cargo_toml_data, &crate_name);
+
+        const DUMMY_ALL_CONTENTS: &str =
+            "# Code generated by the multiversx-sc multi-contract system. DO NOT EDIT.
+
+# ##########################################
```

Comment:
You can place it in a file and load it with `include_file!`, you can get rid of all the escaping.